spec/models/ability_spec.rb

```diff
@@ -25,4 +25,40 @@
       it { is_expected.not_to be_able_to(:update, User.new) }
     end
   end
+
+  describe 'vacations abilities' do
+    describe 'when status is pending' do
+      context 'allows user to cancel a vacation' do
+        it { is_expected.to be_able_to(:destroy, Vacation.new) }
+      end
+    end
+
+    describe 'when status is approved' do
+      let(:vacation) do
+        create(:vacation, {
+            status: :approved,
+            start_date: Date.current.next_day
+          }
+        )
+      end
+
+      it 'does not allow the user to cancel a vacation less than 7 days ahead of time' do
+        allow(Date).to receive(:today).and_return(Date.current - 5)
```

Comment:
Rails has some time helpers for testing code related to dates. Instead of mocking `Date` here we can use `travel_to`. You'll find examples [being used in this project](https://github.com/Codeminer42/Punchclock/search?q=travel_to).

Something like:
```ruby
it 'does not allow the user to cancel a vacation less than 7 days ahead of time' do
  create(:vacation, status: :approved, start_date: Date.tomorrow)

  travel_to 5.days.ago

  is_expected.not_to be_able_to(:destroy, vacation)
end        
```

Ref: https://api.rubyonrails.org/v5.2.4/classes/ActiveSupport/Testing/TimeHelpers.html